runtime/compiler/p/runtime/PPCHWProfilerPrivate.hpp

```diff
@@ -351,21 +351,60 @@ void readBHRB(uintptr_t *bhrb)
 
 #if defined(AIXPPC)
 
-// Defined by later versions of pmapi.h
-#pragma mc_func tr_pmc_read_1to4 { "48008493" }
-#pragma mc_func tr_pmc_read_5to6 { "480084D3" }
-#pragma mc_func tr_mmcr_read { "48008503" }
-#pragma mc_func tr_sampled_regs_read { "48008533" }
-#pragma mc_func tr_pmc_write { "48008563" }
-#pragma mc_func tr_mmcr_write { "480085F3" }
-
-// XXX: However these are defined incorrectly in some versions of pmapi.h
-#pragma reg_killed_by tr_pmc_read_1to4 gr3, gr4, lr
-#pragma reg_killed_by tr_pmc_read_5to6 gr3, gr4, lr
-#pragma reg_killed_by tr_mmcr_read gr3, gr4, lr
-#pragma reg_killed_by tr_sampled_regs_read gr3, gr4, lr
-#pragma reg_killed_by tr_pmc_write gr3, gr5, cr0, lr
-#pragma reg_killed_by tr_mmcr_write gr3, gr5, cr0, lr
+//Milli calls to functions defined by later versions of pmapi.h
+static inline __attribute__((always_inline))
+int tr_pmc_read_1to4(uint32_t buf[4])
+   {
+   register int retval asm("r3");
+   
+   asm("bla %1" : "=r" (retval) : "i" (0x00008490) : "r3", "r4", "lr");
```

Comment:
You should be able to remove "r3" from the clobber list for all cases now since it's bound to `retval`.